File: clients/mobile/lib/utils/security_platform_settings_util.dart
/// Middleware defaults mirrored from `platform/api/src/emcap/auth/middleware.py`.
const securityRateLimitPerMinute = 120;

class SecurityPlatformSettings {
  const SecurityPlatformSettings({
    required this.rateLimitPerMinute,
    required this.securityHeadersEnabled,
    required this.mfaEnrollment,
    required this.abacPolicyCount,
  });

  final int rateLimitPerMinute;
  final bool securityHeadersEnabled;
  final String mfaEnrollment;
  final int abacPolicyCount;
}

/// Read-only security posture from GET /config/platform payload.
SecurityPlatformSettings parseSecurityPlatformSettings(Map<String, dynamic> config) {
  final security = config['security'];
  final policies = security is Map ? security['abac_policies'] : null;
  return SecurityPlatformSettings(
    rateLimitPerMinute: securityRateLimitPerMinute,
    securityHeadersEnabled: true,
    mfaEnrollment: 'account',
    abacPolicyCount: policies is List ? policies.length : 0,
  );
}

File: clients/web/src/app/shared/utils/security-platform-settings.util.ts
/** Middleware defaults mirrored from `platform/api/src/emcap/auth/middleware.py`. */
export const SECURITY_RATE_LIMIT_PER_MINUTE = 120;

export interface SecurityPlatformSettings {
  rateLimitPerMinute: number;
  securityHeadersEnabled: boolean;
  mfaEnrollment: 'account';
  abacPolicyCount: number;
}

/** Read-only security posture from GET /config/platform payload. */
export function parseSecurityPlatformSettings(
  config: Record<string, unknown>,
): SecurityPlatformSettings {
  const security = config['security'] as Record<string, unknown> | undefined;
  const policies = security?.['abac_policies'];
  return {
    rateLimitPerMinute: SECURITY_RATE_LIMIT_PER_MINUTE,
    securityHeadersEnabled: true,
    mfaEnrollment: 'account',
    abacPolicyCount: Array.isArray(policies) ? policies.length : 0,
  };
}
